### PR TITLE
feat: generic validation helper function

### PR DESCRIPTION
**Pull request by jonathangoulding**
check inline error
check summary error

group context pages and assert titles

### CODE REVIEW

File: cypress/integration/internal/sroc-charge-version-flow.spec.js [hunk 1]
```diff
@@ -41,89 +45,223 @@ describe('Create SRoC Charge version workflow journey', () => {
       cy.get('.govuk-button').contains('Set up a new charge').click();
     });
 
-    describe('user selects reason for new charge information', () => {
-      cy.get('.govuk-heading-l').contains('Select reason for new charge information');
-      cy.get('[type="radio"]#reason-12').click();
-      cy.get('button.govuk-button').click();
+    describe('Select reason for new charge information', () => {
```
Comment (jonathangoulding) [inline cypress/integration/internal/sroc-charge-version-flow.spec.js:48]: Describe relates to the page being loaded with a reference to the page title. 

Should allow easier reading, collapsing in a text editor to make the file smaller. 

can be one line:  
validateRadioOptions('Select reason for new charge information', 'reason-12', 'Select a reason for new charge information');

File: cypress/support/validation.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const checkInlineAndSummaryErrorMessage = (expectedMessage) => {
+    cy.get('.govuk-error-summary__list').contains(expectedMessage);
```
Comment (jonathangoulding) [inline cypress/support/validation.js:2]: Screenshot 2022-04-01 at 11 37 52

Comment (jonathangoulding) [reply cypress/support/validation.js:2]: can generically target summary and inline error messages. assuming they are the same as per GDS.

File: cypress/integration/internal/sroc-charge-version-flow.spec.js (hunk 1 above)
Comment (jonathangoulding) [reply cypress/integration/internal/sroc-charge-version-flow.spec.js:48]: https://design-system.service.gov.uk/patterns/question-pages/

All Page titles should be unique. 

Introducing them in this way allows us to establish un expected reuse of the title and to wrap each page to be tested in a describe block with the title as a unique key.

**Review by Stephan-Hough — APPROVED** [review]

**Pull request merged**